Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs

```diff
@@ -262,7 +262,14 @@ private Task ProcessMessageAsync(Node node, IncomingMessage message)
         /// <param name="message">The message payload.</param>
         private async Task SendMempoolPayloadAsync(Node node, MempoolPayload message)
         {
-            Guard.Assert(node == this.AttachedNode); // just in case
+            Guard.NotNull(node, nameof(node));
+            if (node != this.AttachedNode)
+            {
+                this.logger.LogDebug("Node Mismatch ({0}:'{1}', {2}:'{3}')", nameof(node), node.RemoteSocketEndpoint, nameof(this.AttachedNode), this.AttachedNode?.RemoteSocketEndpoint);
+                this.logger.LogTrace("(-)[NODE_MISMATCH]");
+                return;
+            }
+
             this.logger.LogTrace("({0}:'{1}',{2}:'{3}')", nameof(node), node.RemoteSocketEndpoint, nameof(message), message.Command);
```

Comment:
the entry log should be preserved just after guard

then the if is OK, but should not be followed by entry log
it should be followed by log line such as
```
this.logger.LogDebug("Attached node '{0}' does not match the originating node '{1}'.", node.RemoteSocketEndpoint, this.AttachedNode?.RemoteSocketEndpoint);
```

same for the code below